activestorage/app/models/active_storage/preview.rb

```diff
@@ -66,6 +66,12 @@ def url(**options)
     end
   end
 
+  # Returns a combination key of the blob and the variation that together identifies a specific variant.
+  def key
+    variant
+    "variants/#{image.key}/#{Digest::SHA256.hexdigest(variation.key)}"
+  end
```

Comment:
This is incompatible with database-tracked variants. We can implement `ActiveStorage::VariantWithRecord#key` and ask the variant for its key rather than reconstructing it here.

```ruby
# Here:
def key
  if processed?
    variant.key
  else
    raise UnprocessedError
  end
end
```

```ruby
# lib/active_storage/variant_with_record.rb
def key
  image&.key
end
```